File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptArrayOptStringValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptArrayOptStringInValue.OptArrayOptStringValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptArrayOptStringValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptArrayOptStringValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                String[].class,
                OptArrayOptStringValue.class.getDeclaredField("optArrayOptString").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                String[].class,
                OptArrayOptStringValue.class.getMethod("getOptArrayOptString").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptArrayOptStringValue.class,
                OptArrayOptStringValue.class.getMethod("setOptArrayOptString", String[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptArrayOptStringValue().getOptArrayOptString());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptArrayOptStringValue.scala
package com.dslplatform.ocd.values

import OptArrayOptStringInValue.OptArrayOptStringValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptArrayOptStringValue extends SpecExtensions {
  def `Test property field type: Option[Array[Option[String]]]`() =
    checkType(
      typeOf[Option[Array[Option[String]]]]
    , typeOf[OptArrayOptStringValue].member("optArrayOptString": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptArrayOptStringValue().optArrayOptString === None)
  }
}
